NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt

```diff
@@ -0,0 +1,226 @@
+import androidx.compose.ui.test.onNodeWithContentDescription
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performGesture
+import androidx.compose.ui.test.swipe
+import androidx.compose.ui.test.swipeDown
+import androidx.compose.ui.test.topCenter
+import androidx.compose.ui.unit.ExperimentalUnitApi
+import androidx.compose.ui.unit.dp
+import com.microsoft.device.display.samples.navigationrail.models.DataProvider.plantList
+import com.microsoft.device.display.samples.navigationrail.ui.components.DrawerState
+import com.microsoft.device.display.samples.navigationrail.ui.components.DrawerStateKey
+import com.microsoft.device.display.samples.navigationrail.ui.theme.NavigationRailAppTheme
+import com.microsoft.device.display.samples.navigationrail.ui.view.ItemDetailView
+import com.microsoft.device.display.samples.navigationrail.ui.view.NavigationRailApp
+import com.microsoft.device.display.samples.navigationrail.ui.view.Pane2
+import com.microsoft.device.dualscreen.testutils.getString
+import com.microsoft.device.dualscreen.windowstate.WindowState
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalFoundationApi
+@ExperimentalUnitApi
+@ExperimentalMaterialApi
+@ExperimentalAnimationApi
+class DetailTest {
+    @get: Rule
+    val composeTestRule = createAndroidComposeRule<MainActivity>()
+
+    /**
+     * Tests that the back button appears in pane 2 in single screen mode
+     */
+    @Test
+    fun pane2_backButtonAppearsInSingleScreenMode() {
+        composeTestRule.setContent {
+            Pane2Plants(isDualPortrait = false, isDualLandscape = false)
+        }
+
+        // Assert that back button is visible
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back)).assertIsDisplayed()
+    }
+
+    /**
+     * Tests that the back button appears in pane 2 in dual landscape mode
+     */
+    @Test
+    fun pane2_backButtonAppearsInDualLandscapeMode() {
+        composeTestRule.setContent {
+            Pane2Plants(isDualPortrait = false, isDualLandscape = true)
+        }
+
+        // Assert that back button is visible
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back)).assertIsDisplayed()
+    }
+
+    /**
+     * Tests that the back button doesn't appear in pane 2 in dual portrait mode
+     */
+    @Test
+    fun pane2_backButtonHiddenInDualPortraitMode() {
+        composeTestRule.setContent {
+            Pane2Plants(isDualPortrait = true, isDualLandscape = false)
+        }
+
+        // Assert that back button does not exist
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back)).assertDoesNotExist()
+    }
+
+    /**
+     * Tests that the back button successfully returns to the gallery view from the item detail view
+     */
+    @Test
+    fun app_backButtonChangesPanes() {
```

Comment:
Can we rename this by adding "InSingleScreenMode" and the comment as well?

Reply:
Not too long at all, good catch!